core/object.go

```diff
@@ -1356,7 +1360,7 @@ func (rx Regex) Print(w io.Writer, printReadably bool) {
 func (rx Regex) Equals(other interface{}) bool {
 	switch other := other.(type) {
 	case Regex:
-		return rx.R == other.R
+		return rx.R == other.R || rx.R.String() == other.R.String()
```

Comment:
This is not how regex equality works in Clojure:
```
user=> (= #"asdf" #"asdf")
false
```
So I think the current version is correct.

Reply:
> This is not how regex equality works in Clojure:
> 
> ```
> user=> (= #"asdf" #"asdf")
> false
> ```
> 
> So I think the current version is correct.

Oops, didn't think to check that! Fixed.

And this led me to notice that `(joker.string/re-string r)` is just another way of saying `(str r)`, i.e. Joker users already have that function.

So that leaves only `joker.string/re-quoted` as new functionality here (plus the tests).

Is it still worth introducing `joker.string/re-string`? If not, should `re-quoted` be moved back into `joker.core` for consistency with the others?